### PR TITLE
Reclassification Level Triggers & Sysadmin Conditional Triggers

### PR DESCRIPTION
**Pull request by JVickery-TBS**
fix(pd): triggers;

- Fix reclass level triggers.
- Fix sysadmin username triggers.

### CODE REVIEW

File: ckanext/canada/triggers.py [hunk 1]
```diff
@@ -280,19 +282,21 @@ def update_triggers():
                 IF NEW.record_created IS NULL THEN
                     NEW.record_created := OLD.record_created;
                 END IF;
-                IF NEW.record_modified IS NULL THEN
+                IF NOT sysadmin AND NEW.record_modified IS NULL THEN
                     NEW.record_modified := OLD.record_modified;
                 END IF;
-                IF (req_user_modified = '') IS NOT FALSE THEN
+                IF NOT sysadmin AND (req_user_modified = '') IS NOT FALSE THEN
```
Comment (wardi) [inline ckanext/canada/triggers.py:288]: This shouldn't be necessary. `req_user_modified` is forced to NULL when NOT sysadmin, only sysadmins may set a user_modified value.

File: ckanext/canada/triggers.py [hunk 2]
```diff
@@ -280,19 +282,21 @@ def update_triggers():
...
                     NEW.user_modified := OLD.user_modified;
                 END IF;
                 IF OLD = NEW THEN
                     RETURN NULL;
                 END IF;
-                NEW.record_modified := now() at time zone 'utc';
-                IF (req_user_modified = '') IS NOT FALSE THEN
+                IF NOT sysadmin THEN
```
Comment (wardi) [inline ckanext/canada/triggers.py:294]: I think we need to record that a sysadmin passed a value for `record_modified` the way we did for `req_user_modified` above so that we can keep that value here instead of overwriting it. That way when sysadmins don't pass a `record_modified` the value will be updated here as normal.

**Comment by wardi** [comment]
so many edge cases... really could use some tests to make sense of this one

**Review by wardi — APPROVED** [review]

File: ckanext/canada/tests/test_triggers.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        # sysadmin upserts should not modify these values
+        assert record['record_created'] == current_created_time
+        assert record['user_modified'] == current_user_modified
+        assert record['record_modified'] == current_modified_time
+
+        record['summary_en'] = 'Even Newer English Summary'
+        record['summary_fr'] = 'Even Newer French Summary'
+
+        # need to wait to upsert to make sure modified timestamps change or not
+        sleep(5)
```
[the remark this answers is not in the document]
Comment (JVickery-TBS) [reply ckanext/canada/tests/test_triggers.py:98]: @wardi at least in my docker container:
```
AssertionError: assert '2024-04-19T20:06:40' != '2024-04-19T20:06:40'
```

[the remark this answers is not in the document]
Comment (wardi) [reply ckanext/canada/tests/test_triggers.py:98]: interesting, we shouldn't be truncating timestamps on output that much in datastore, but that's a different issue. 1s delay should be sufficient for the test.

[the remark this answers is not in the document]
Comment (wardi) [reply ckanext/canada/tests/test_triggers.py:98]: or you could probably get exact values with the right datastore_search_sql incantation
